File: test/net/dinkla/raytracer/objects/utilities/BinaryPlyReaderTest.java
package net.dinkla.raytracer.objects.utilities;

import net.dinkla.raytracer.TestUtils;
import net.dinkla.raytracer.objects.acceleration.Grid;
import net.dinkla.raytracer.objects.mesh.Mesh;
import org.junit.Test;

import static org.junit.Assert.assertEquals;

public class BinaryPlyReaderTest {

//    @Test
    public void ReadFlat() {
        Mesh mesh = new Mesh();
        Grid grid = new Grid(mesh);
        BinaryPlyReader.read(grid, TestUtils.PLY_EXAMPLE, false, false);
        //System.out.println("mesh=" + grid.getMesh());
        assertEquals(grid.getMesh().getVertices().size(), 4);
        assertEquals(grid.size(), 2);
    }

//    @Test
    public void ReadSmooth() {
        Mesh mesh = new Mesh();
        Grid grid = new Grid(mesh);
        BinaryPlyReader.read(grid, TestUtils.PLY_EXAMPLE, false, true);
        //System.out.println("mesh=" + grid.getMesh());
        assertEquals(grid.getMesh().getVertices().size(), 4);
        assertEquals(grid.size(), 2);
    }

}

File: src/test/kotlin/net/dinkla/raytracer/objects/utilities/BinaryPlyReaderTest.kt
package net.dinkla.raytracer.objects.utilities

import net.dinkla.raytracer.PLY_EXAMPLE
import net.dinkla.raytracer.objects.acceleration.Grid
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Disabled
import org.junit.jupiter.api.Test

class BinaryPlyReaderTest {

    @Disabled
    @Test
    fun readFlat() {
        val grid = Grid()
        BinaryPlyReader.read(grid, PLY_EXAMPLE, reverseNormal = false, isSmooth = false)
        //System.out.println("mesh=" + grid.getMesh());
        assertEquals(grid.mesh.vertices.size, 4)
        assertEquals(grid.size(), 2)
    }

    @Disabled
    @Test
    fun readSmooth() {
        val grid = Grid()
        BinaryPlyReader.read(grid, PLY_EXAMPLE, reverseNormal = false, isSmooth = true)
        //System.out.println("mesh=" + grid.getMesh());
        assertEquals(grid.mesh.vertices.size, 4)
        assertEquals(grid.size(), 2)
    }

}
